src/System.Runtime.InteropServices.RuntimeInformation/src/OSPlatform.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSPlatform : IEquatable<OSPlatform>
+    {
+        private readonly string _osPlatformString;
+
+        private const string WindowsString = "WINDOWS";
+        private const string LinuxString = "LINUX";
+        private const string OSXString = "OSX";
+
+        public static OSPlatform Windows
+        {
+            get
+            {
+                return new OSPlatform(WindowsString);
```

Comment:
In the API review, the guidance was to make these static readonly properties instead of static readonly fields, to allow for more flexibility for the future (e.g. to allow lazy initialization, etc.).

The way its currently implemented means the constructor will be called with two `if` checks every time the property is accessed.

Instead, maybe just pre-create these three values and save them in private static readonly fields to avoid the constructor calls each time the property is accessed? Or, keep the implementation of the properties as-is but move the if/throw checks to the factory method and make the constructor private with `Debug.Assert`s for the constructor's argument checks.